instrumentation/camel-2.20/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/AwsSpanAssertions.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apachecamel.aws;
+
+import static io.opentelemetry.api.common.AttributeKey.longKey;
+import static io.opentelemetry.api.common.AttributeKey.stringKey;
+import static io.opentelemetry.api.trace.SpanKind.CLIENT;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.equalTo;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.satisfies;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.sdk.testing.assertj.SpanDataAssert;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+
+class AwsSpanAssertions {
+
+  private AwsSpanAssertions() {}
+
+  static void sqs(SpanDataAssert span, int index, String spanName) {
+    sqs(span, index, spanName, null, null, CLIENT, null);
+  }
+
+  static void sqs(SpanDataAssert span, int index, String spanName, String queueUrl) {
+    sqs(span, index, spanName, queueUrl, null, CLIENT, null);
+  }
+
+  static void sqs(
+      SpanDataAssert span, int index, String spanName, String queueUrl, String queueName) {
+    sqs(span, index, spanName, queueUrl, queueName, CLIENT, null);
+  }
+
+  public static void sqs(
+      SpanDataAssert span,
+      int index,
+      String spanName,
+      String queueUrl,
+      String queueName,
+      SpanKind spanKind) {
+    sqs(span, index, spanName, queueUrl, queueName, spanKind, null);
+  }
+
+  static void sqs(
+      SpanDataAssert span,
+      int index,
+      String spanName,
+      String queueUrl,
+      String queueName,
+      SpanKind spanKind,
+      SpanData parentSpan) {
+    if (index == 0) {
+      span.hasNoParent();
+    } else {
+      span.hasParent(parentSpan);
+    }
```

Comment:
The value of `index` isn't really used anywhere. You could check whether `parentSpan == null` instead.

Or, even better: instead of `void` return a `SpanDataAssert` and make it possible to chain the assertions, e.g.
```java
sqs(span, ...).hasParent(trace.getSpan(0))
```
You could get rid of some of these overloads in the preceding lines.

Same for all other assertion methods: you could return the `SpanDataAssert` instead if using `index` (or any other groovy things)

Reply:
great idea! I updated this